### PR TITLE
Course activation basics - PMT #105457

### PR DESCRIPTION
**Pull request by nikolas**
These features are behind the 'instructor_homepage' waffle flag

### CODE REVIEW

File: mediathread/main/views.py [hunk 1]
```diff
@@ -723,6 +733,30 @@ class HomepageView(LoggedInAsFacultyMixin, CourseListView):
 
     def get_context_data(self, **kwargs):
         context = super(HomepageView, self).get_context_data(**kwargs)
-        activatable_courses = []
+        if not waffle.flag_is_active(self.request, 'instructor_homepage'):
+            return context
+
+        groups = self.request.user.groups.all()
```
Comment (sdreher) [inline mediathread/main/views.py]: Picking up all groups will also include non-course based groups associated with collaborations. There's some complexity here as well as to what groups are created for the user at login. Maybe we can walk through this flow together?

File: mediathread/main/views.py [hunk 2]
```diff
@@ -72,6 +78,10 @@ def triple_homepage(request):
     if not request.course:
         return HttpResponseRedirect('/accounts/login/')
 
+    if waffle.flag_is_active(request, 'instructor_homepage') and \
```
Comment (sdreher) [inline mediathread/main/views.py]: Placing the redirect here appears to make it impossible to login to a course.

File: mediathread/main/views.py (hunk 1 above)
Comment (nikolas) [inline mediathread/main/views.py]: Yeah.. I wanted to talk to you about this, let me know when you have a chance.

File: mediathread/main/views.py (hunk 2 above)
Comment (nikolas) [inline mediathread/main/views.py]: I'll fix this and come up with a test for that.

**Pull request closed without merging**